chain-core/src/state/account.rs

```diff
@@ -186,14 +194,41 @@ impl StakedStateOpAttributes {
 
 /// takes UTXOs inputs, deposits them in the specified StakedState's bonded amount - fee
 /// (updates StakedState's bonded + nonce)
-#[derive(Debug, PartialEq, Eq, Clone, Encode, Decode)]
+#[derive(Debug, PartialEq, Eq, Clone, Encode)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct DepositBondTx {
     pub inputs: Vec<TxoPointer>,
     pub to_staked_account: StakedStateAddress,
     pub attributes: StakedStateOpAttributes,
 }
 
+impl Decode for DepositBondTx {
+    fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
+        let size = input
+            .remaining_len()?
+            .ok_or_else(|| "Unable to calculate size of input")?;
+
+        if size > MAX_DEPOSIT_TX_SIZE {
```

Comment:
because utxo table is vector, it can be filered prior to deserialization

Reply:
what do you think ?  @tomtau

Reply:
do you mean to find the length as an offset in the binary?

Reply:
yes,
if vector is really big, 
total message size is also big,
then it will be already filtered out.

Reply:
if we apply limit of individual vectors,
then program can be more complicated.